staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -15,9 +15,38 @@
 package model
 
 import (
+	"time"
+
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
 	"k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
 )
 
+const (
+	// the largest request that will be accepted is 3MB
+	// TODO(DangerOnTheRanger): wire in MaxRequestBodyBytes from apiserver/pkg/server/options/server_run_options.go to make this configurable
+	maxRequestSizeBytes = int64(3 * 1024 * 1024)
+	// OpenAPI duration strings follow RFC 3339, section 5.6 - see the comment on maxDatetimeSizeJSON
+	maxDurationSizeJSON = 32
+	// OpenAPI datetime strings follow RFC 3339, section 5.6, and the longest possible
+	// such string is 9999-12-31T23:59:59.999999999Z, which has length 30 - we add 2
+	// to allow for quotation marks
+	maxDatetimeSizeJSON = 32
+	// Golang allows a string of 0 to be parsed as a duration, so that plus 2 to account for
+	// quotation marks makes 3
+	minDurationSizeJSON = 3
+	// RFC 3339 dates require YYYY-MM-DD, and then we add 2 to allow for quotation marks
+	dateSizeJSON = 12
+	// RFC 3339 times require 2-digit 24-hour time at the very least plus a capital T at the start,
+	// e.g., T23, and we add 2 to allow for quotation marks as usual
+	minTimeSizeJSON = 5
+	// RFC 3339 datetimes require a full date (YYYY-MM-DD) and full time (HH:MM:SS), and we add 3 for
+	// quotation marks like always in addition to the capital T that separates the date and time
+	minDatetimeSizeJSON = 21
+	// The largest 64-bit signed integer can have 19 digits, and we add 2 to account for quotation marks
+	maxNumberSizeJSON = 21
+)
+
 // SchemaDeclType converts the structural schema to a CEL declaration, or returns nil if the
 // the structural schema should not be exposed in CEL expressions.
 // Set isResourceRoot to true for the root of a custom resource or embedded resource.
```

Comment:
if the user sets explicit maxLength / maxItems / maxProperties that *exceed* the actual size / number of items they could fit in the request, I think we should take their higher value, since we want to discourage such excessive maximums

Reply:
The penalty of having higher costs if excessive limits are set sounds right to me too. It's also easier for us to document the how cost calculation work if we always use a provided limit.

Reply:
Right - I've gone ahead and removed `maxNumberSizeJSON` and related code.